climakitaegui/explore/amy.py

```diff
@@ -429,68 +430,79 @@ def _update_tmy_computatation_description(self):
-                    cached_area_str, self.computation_method, self.warmlevel
-                )
-                clabel = self.variable + " (" + self.units + ")"
-            else:
-                df = compute_severe_yr(
-                    self.future_tmy_data, days_in_year=days_in_year
-                ) - compute_amy(self.historical_tmy_data, days_in_year=days_in_year)
-                title = "Severe Meteorological Year: {}\nDifference between {} at 90th percentile and Historical Baseline".format(
-                    cached_area_str, self.computation_method
-                )
-                clabel = self.variable + " (" + self.units + ")"
-        else:
-            title = "Average Meteorological Year for\n{}".format(cached_area_str)
+        match self.amy_type:
+            case "Absolute":
+                if self.computation_method == "Historical":
+                    df = compute_amy(
+                        self.historical_tmy_data, days_in_year=days_in_year
+                    )
+                    title = (
+                        "Average Meteorological Year: {}\nAbsolute {} Baseline".format(
+                            cached_area_str, self.computation_method
+                        )
+                    )
+                    clabel = (
+                        self.variable
+                        + " ("
+                        + self.historical_tmy_data.attrs["units"]
+                        + ")"
+                    )
+                else:
+                    df = compute_amy(self.future_tmy_data, days_in_year=days_in_year)
+                    title = (
+                        "Average Meteorological Year: {}\nAbsolute {} at {}°C".format(
+                            cached_area_str, self.computation_method, self.warmlevel
+                        )
+                    )
+                    clabel = self.variable + " (" + self.units + ")"
+            case "Difference":
+                cmap = read_ae_colormap("ae_diverging", cmap_hex=True)
+                if self.computation_method == "Warming Level Future":
+                    df = compute_amy(
+                        self.future_tmy_data, days_in_year=days_in_year
+                    ) - compute_amy(self.historical_tmy_data, days_in_year=days_in_year)
+                    title = "Average Meteorological Year: {}\nDifference between {} at {}°C and Historical Baseline".format(
+                        cached_area_str, self.computation_method, self.warmlevel
+                    )
+                    clabel = self.variable + " (" + self.units + ")"
+                else:
+                    df = compute_severe_yr(
+                        self.future_tmy_data, days_in_year=days_in_year
+                    ) - compute_amy(self.historical_tmy_data, days_in_year=days_in_year)
+                    title = "Severe Meteorological Year: {}\nDifference between {} at 90th percentile and Historical Baseline".format(
```

Comment:
df computation can be moved up and out of `if else` statement, clabel can be moved down and out of `if else`

Reply:
I can see moving clabel down and out of `if`. Not sure about `df` compute, one is `compute_amy` and the other is `compute_severe_yr`. Can you explain this more?